src/Columns/ColumnVector.h

```diff
@@ -46,7 +46,22 @@ struct CompareHelper
       */
     static constexpr int compare(T a, U b, int /*nan_direction_hint*/)
     {
-        return a > b ? 1 : (a < b ? -1 : 0);
+        if constexpr (std::endian::native == std::endian::big && std::same_as<T, UUID> && std::same_as<U, UUID>)
+        {
+            UUID tmp_a = a;
```

Comment:
See also `memcmp`.

Reply:
I changed the code to use memcmp, but we still need to reverse byte orders before using memcmp.

Reply:
That's strange, I expect we can simplify further.
Maximum - exchange of two halves and/or memcmp.

Reply:
It seems like an utility function "reverseMemcmp()" is needed, it can be used to simplify the code.